hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALKey.java

```diff
@@ -57,6 +57,14 @@ default long estimatedSerializedSizeOf() {
    */
   TableName getTableName();
 
+  /**
+   * set the table name for WALKey. This is mainly for the purpose of
+   * when we want to change the table name in the WALKey.
+   * For example:
+   * replicating across different table names in two different clusters
+   */
+  void setTableName(TableName tableName);
```

Comment:
WALKey only has getters. It doesn't say it is immutable in the class comment but it should.